bundles/org.openhab.binding.modbus.e3dc/src/main/java/org/openhab/binding/modbus/e3dc/internal/handler/E3DCWallboxThingHandler.java

```diff
@@ -0,0 +1,261 @@
+import org.openhab.io.transport.modbus.ModbusWriteRegisterRequestBlueprint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link E3DCWallboxThingHandler} Basic modbus connection towards the E3DC device
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class E3DCWallboxThingHandler extends BaseThingHandler {
+    public enum ReadWriteSuccess {
+        NOT_RECEIVED,
+        SUCCESS,
+        FAILED
+    }
+
+    private static final String READ_WRITE_ERROR = "Modbus Data Read/Write Error";
+    private static final String READ_ERROR = "Modbus Read Error";
+    private static final String WRITE_ERROR = "Modbus Write Error";
+
+    ChannelUID wbAvailableChannel;
+    ChannelUID wbSunmodeChannel;
+    ChannelUID wbChargingAbortedChannel;
+    ChannelUID wbChargingChannel;
+    ChannelUID wbJackLockedChannel;
+    ChannelUID wbJackPluggedChannel;
+    ChannelUID wbSchukoOnChannel;
+    ChannelUID wbSchukoPluggedChannel;
+    ChannelUID wbSchukoLockedChannel;
+    ChannelUID wbSchukoRelay16Channel;
+    ChannelUID wbRelay16Channel;
+    ChannelUID wbRelay32Channel;
+    ChannelUID wb1phaseChannel;
+
+    private final Logger logger = LoggerFactory.getLogger(E3DCWallboxThingHandler.class);
+    private final Parser dataParser = new Parser(DataType.DATA);
+    private ReadWriteSuccess dataRead = ReadWriteSuccess.NOT_RECEIVED;
+    private ReadWriteSuccess dataWrite = ReadWriteSuccess.NOT_RECEIVED;
+    private volatile BitSet currentBitSet = new BitSet(16);
+    private @Nullable E3DCWallboxConfiguration config;
+    private @Nullable E3DCThingHandler bridgeHandler;
+
+    public E3DCWallboxThingHandler(Thing thing) {
+        super(thing);
+        wbAvailableChannel = new ChannelUID(thing.getUID(), WB_AVAILABLE_CHANNEL);
+        wbSunmodeChannel = new ChannelUID(thing.getUID(), WB_SUNMODE_CHANNEL);
+        wbChargingAbortedChannel = new ChannelUID(thing.getUID(), WB_CHARGING_ABORTED_CHANNEL);
+        wbChargingChannel = new ChannelUID(thing.getUID(), WB_CHARGING_CHANNEL);
+        wbJackLockedChannel = new ChannelUID(thing.getUID(), WB_JACK_LOCKED_CHANNEL);
+        wbJackPluggedChannel = new ChannelUID(thing.getUID(), WB_JACK_PLUGGED_CHANNEL);
+        wbSchukoOnChannel = new ChannelUID(thing.getUID(), WB_SCHUKO_ON_CHANNEL);
+        wbSchukoPluggedChannel = new ChannelUID(thing.getUID(), WB_SCHUKO_PLUGGED_CHANNEL);
+        wbSchukoLockedChannel = new ChannelUID(thing.getUID(), WB_SCHUKO_LOCKED_CHANNEL);
+        wbSchukoRelay16Channel = new ChannelUID(thing.getUID(), WB_SCHUKO_RELAY_16A_CHANNEL);
+        wbRelay16Channel = new ChannelUID(thing.getUID(), WB_RELAY_16A_CHANNEL);
+        wbRelay32Channel = new ChannelUID(thing.getUID(), WB_RELAY_32A_CHANNEL);
+        wb1phaseChannel = new ChannelUID(thing.getUID(), WB_1PHASE_CHANNEL);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(() -> {
```

Comment:
Is there a reason why you execute this asynchronously?

Reply:
It's the pattern generated by the binding skeleton script. But of course there are no extensive actions. I removed scheduler and get config and bridge synchronous.